### PR TITLE
Fixes to the Number Input

### LINKED ISSUE #1665: Decimal steps are not working in the Number Input
**Describe the bug**
Setting a decimal step on the Number Input is not working as expected. Only the first click works decently, the followings keep the same value and doesn't update.

**To Reproduce**
- Create a React App with the following code:
```
() => {
  const [value, setValue] = useState("");
  const onChange = ({ value }) => {
    setValue(value);
  };
  const onBlur = ({ value }) => {
    setValue(value);
  };

  return (
    <DxcInset space="2rem">
      <DxcNumberInput
        label="Enter your age"
        value={value}
        onChange={onChange}
        onBlur={onBlur}
        step={0.01}
      />
    </DxcInset>
  );
}
```

**Expected behaviour**
Update the value with the correct decimal step.

### PR DESCRIPTION
**Pull request by GomezIvann**
**Checklist**
- [x] Build process is done without errors. All tests pass in the `/lib` directory.
- [x] Self-reviewed the code before submitting.
- [x] Meets accessibility standards.
- [x] Added/updated documentation to `/website` as needed.
- [x] Added/updated tests as needed.

**Description**
Some updates were made to make the Number input behave as its native counterpart. Increment and decrement actions replicate the spin buttons of the HTML number input. New tests were added for those edge cases. Also, a new feature has been added: the `step` can now have decimals.

All these fixes required some updates in the Text Input, which also led to needed changes on the components that use it under the hood, like the Password input.

**Closes #1665**

### CODE REVIEW

File: lib/src/password-input/PasswordInput.test.js [hunk 1]
```diff
@@ -79,15 +75,14 @@ describe("Password input component tests", () => {
...
     const { getByRole, getByLabelText } = render(<DxcPasswordInput label="Password input" />);
     const passwordInput = getByLabelText("Password input");
-    expect(passwordInput.getAttribute("aria-autocomplete")).toBeNull();
-    expect(passwordInput.getAttribute("aria-controls")).toBeNull();
-    expect(passwordInput.getAttribute("aria-expanded")).toBeNull();
     const showButton = getByRole("button");
+    expect(passwordInput.getAttribute("aria-expanded")).toBe("false");
     expect(showButton.getAttribute("aria-label")).toBe("Show password");
     userEvent.click(showButton);
+    expect(passwordInput.getAttribute("aria-expanded")).toBe("true");
```
[the remark this answers is not in the document]
Comment (GomezIvann) [reply lib/src/password-input/PasswordInput.test.js:85]: I remove the checks of the `aria-autocomplete`, `aria-controls` since they were part of the behaviour of the regular Text Input and they are checked in its respective tests. But, I don't have any problem in restoring them. 

In regards to the `aria-expanded`, I think its value should only be true or false, depending on the visibility of the password text. So, I propose to keep it as it is.

[the remark this answers is not in the document]
Comment (raquelarrojo) [reply lib/src/password-input/PasswordInput.test.js:85]: Agree!

**Review by raquelarrojo — APPROVED** [review]
